src/Combinatorics/Matroids/properties.jl

```diff
@@ -1030,6 +1030,58 @@ function min_revlex_basis_encoding(M::Matroid)
     return _revlex_basis_from_vector(min_rvlx)
 end
 
+@doc raw"""
+    matroid_hex(M::Matroid)
+
+Stores a matroid as a string of hex characters. The first part of the string is "r" followed by the rank of the matroid. This is followed by "n" and the number of elements. The rest of the string is the revlex basis encoding. The encoding is done by converting the basis encoding to a vector of bits and then to a string of characters. The bits are padded to a multiple of 4 and then converted to hex characters.
+
+# Examples
+To get the hex encoding of the fano matroid write:
+```jldoctest
+julia> matroid_hex(fano_matroid())
+"r3n7_3f7eefd6f"
+
+```
+"""
+function matroid_hex(M::Matroid)::String
+  rvlx = min_revlex_basis_encoding(M)
+  r,n = rank(M), length(M) 
+
+  v = _revlex_basis_to_vector(rvlx)
+  for _ in 1:(4-length(v)%4)
+    pushfirst!(v,0)
+  end
+  v = reshape(v,4,:)
+  v = [string(parse(Int,join(v[:,j]),base=2),base=16) for j in 1:size(v)[2]]
+
+  return "r$(r)n$(n)_" * join(v)
+end
+
+@doc raw"""
+    matroid_from_matroid_hex(str::AbstractString)
+
+Returns a matroid from a string of hex characters.
+
+# Examples
+To retrieve the fano matroid from its hex encoding write:
+
+```jldoctest
+julia> matroid_from_matroid_hex("r3n7_3f7eefd6f")
+Matroid of rank 3 on 7 elements
+
+```
+"""
+function matroid_from_matroid_hex(str::AbstractString)::Matroid
+  sep = split(str,"_")
+  (r,n) = parse.(Int,split(sep[1][2:end],"n"))
```

Comment:
Have you thought of this? not sure which is better, I do think it might be a little easier to recognize whats going on here with regex.
```suggestion
  (r, n) = string.(match(r"r([0-9A-Fa-f]+)n([0-9A-Fa-f]+)").captures)
```

Reply:
I have thought of this but I believe it's slower.

Reply:
also, there seems to be no error handling, this function should throw an error when the string isn't valid

Reply:
I have added some error handling to this function. But I can only check if the syntax is correct, it is impossible to check if its a matroid without constructing it.